src/shapes/ISO_enter.scad

```diff
@@ -32,3 +33,23 @@ module ISO_enter_shape(size, delta, progress){
     }
   }
 }
+
+function iso_enter_vertices(width, height, width_ratio, height_ratio, wd, hd) = [
+  [                   0-wd,                     0-hd], // top right
+  [                   0-wd,               -height+hd], // bottom right
+  [-width * width_ratio+wd,               -height+hd], // bottom left
+  [-width * width_ratio+wd,-height * height_ratio+hd], // inner middle point
+  [              -width+wd,-height * height_ratio+hd], // outer middle point
+  [              -width+wd,                     0-hd]  // top left
+] + [
+  [(width * width_ratio)/2, height/2 ],
+  [(width * width_ratio)/2, height/2 ],
+  [(width * width_ratio)/2, height/2 ],
+  [(width * width_ratio)/2, height/2 ],
+  [(width * width_ratio)/2, height/2 ],
+  [(width * width_ratio)/2, height/2 ]
+];
+
+// no rounding on the corners at all
+function skin_iso_enter_shape(size, delta, progress, thickness_difference) =
```

Comment:
passing thickness_difference is because we've been assuming convex shapes where we are in the center of the shape. without passing the thickness difference the "elbow" of the keycap comes out wrong. The center of the keycap is actually below the elbow, so if you have a smaller keyshape, the elbow is closer to the center. In order to generate a keycap we difference a larger keyshape from a smaller one, so if you don't make any accomodations the smaller keycap's elbow will cut through the larger one.

The solution is to pull out the thickness difference you're trying to apply to the keycap, so the shape can do with it what it must, which is mostly just carefully making sure you add/subtract the thickness difference such that the resulting keycap is contained by the larger one.